src/main/java/de/komoot/photon/elasticsearch/Searcher.java

```diff
@@ -57,12 +71,32 @@ public Searcher(Client client) {
 		if(lon != null) params.put("lon", lon);
 		if(lat != null) params.put("lat", lat);
 
-		StrSubstitutor sub = new StrSubstitutor(params.build(), "${", "}");
-		if(lon != null && lat != null) {
-			query = sub.replace(queryLocationBiasTemplate);
-		} else {
-			query = sub.replace(queryTemplate);
-		}
+        this.setupSearch(lon,lat,tagKey,tagValue);
```

Comment:
probably this battery of if/else statements could be replaced by using a template engine and only one json template